### PR TITLE
Revert AuthenticationAddXyz overload changes

### LINKED ISSUE #24048: Fix regression in auth options registration singleton->transient
For context see: https://github.com/dotnet/aspnetcore/commit/1f76cce14ac4e4698a554b65a24f28000b50396e#r40670134

From this PR https://github.com/dotnet/aspnetcore/pull/19268

### PR DESCRIPTION
**Pull request by HaoK**
Fixes https://github.com/dotnet/aspnetcore/issues/24048

### CODE REVIEW

**Comment by Tratcher** [comment]
Preview 8?

File: src/Security/Authentication/Cookies/samples/CookieSessionSample/Startup.cs [hunk 1]
```diff
@@ -14,14 +14,12 @@ public class Startup
...
         {
-            services.AddSingleton<MemoryCacheTicketStore>();
-
             // This can be removed after https://github.com/aspnet/IISIntegration/issues/371
             services.AddAuthentication(options =>
             {
                 options.DefaultAuthenticateScheme = CookieAuthenticationDefaults.AuthenticationScheme;
                 options.DefaultChallengeScheme = CookieAuthenticationDefaults.AuthenticationScheme;
-            }).AddCookie<MemoryCacheTicketStore>((o, ticketStore) => o.SessionStore = ticketStore);
+            }).AddCookie(o => o.SessionStore = new MemoryCacheTicketStore());
```
Comment (Tratcher) [inline src/Security/Authentication/Cookies/samples/CookieSessionSample/Startup.cs:22]: Can you show the workaround here

Comment (HaoK) [reply src/Security/Authentication/Cookies/samples/CookieSessionSample/Startup.cs]: Sure, it just is using the options builder api instead

**Review by Tratcher — APPROVED** [review]
This should get rebased into preview8. Better to make a breaking change before RC. It will need a breaking change announcement as well, people have already started using these.

**Comment by HaoK** [comment]
Okay I'll create a new preview8 PR

**Pull request closed without merging**